### PR TITLE
Edit review

### PR DESCRIPTION
**Pull request by hieuhtk58**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -23,4 +34,9 @@ def review_params
     def load_book
       @book = Book.find_by id: params[:book_id]
     end
+
+    def load_review
+      @review = Review.find(params[:id])
```
Comment (phanbt58) [inline app/controllers/reviews_controller.rb]: sử dụng find_by, remove ()

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -14,6 +14,17 @@ def create
     end
   end
 
+  def edit
+  end
+
+  def update
+    if @review.update(review_params)
```
Comment (phanbt58) [inline app/controllers/reviews_controller.rb]: remove ()

File: config/routes.rb [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
   resources :users
   resources :books, only: [:index, :show] do
-    resources :reviews, only: [:create, :new]
+    resources :reviews, only: [:create, :new, :edit, :update]
```
Comment (phanbt58) [inline config/routes.rb]: chuyển sang dùng expect

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -8,12 +9,23 @@ def new
...
+
+  def update
+    if @review.update review_params
+      redirect_to @book
+    else
+      render :edit
+    end
+  end
+
   private
```
Comment (phanbt58) [inline app/controllers/reviews_controller.rb]: convention

**Comment by ngotungson** [comment]
OK

File: app/controllers/reviews_controller.rb (hunk 4 above)
Comment (phanbt58) [inline app/controllers/reviews_controller.rb]: convention

File: app/controllers/reviews_controller.rb [hunk 5]
```diff
@@ -8,19 +9,34 @@ def new
...
+    if @review.update review_params
+      redirect_to @book
+    else
+      render :edit
     end
+  end
+
+  private
+  def review_params
+      params.require(:review).permit :rating, :content
```
Comment (phanbt58) [inline app/controllers/reviews_controller.rb]: ident

File: app/controllers/reviews_controller.rb [hunk 6]
```diff
@@ -8,19 +9,34 @@ def new
...
+  end
 
-    def load_book
+  def load_book
       @book = Book.find_by id: params[:book_id]
```
Comment (phanbt58) [inline app/controllers/reviews_controller.rb]: ident

File: app/controllers/reviews_controller.rb [hunk 7]
```diff
@@ -8,19 +9,34 @@ def new
...
-    end
+  end
+
+  def load_review
+      @review = Review.find_by id: params[:id]
```
Comment (phanbt58) [inline app/controllers/reviews_controller.rb]: ident

**Comment by phanbt58** [comment]
ok

File: app/views/reviews/_review.html.erb [hunk 8]
```diff
@@ -1,3 +1,4 @@
-<%= review.rating %>
-<%= review.content %>
-<br>
+<div><%= review.rating %></div>
+<div><%= review.content %></div>
+<div><%= link_to t("views.book.show.edit"),
+  edit_book_review_path(review.book, review) %></div>
```
Comment (nhunght) [inline app/views/reviews/_review.html.erb]:
```
<div>
    <%= link_to t("views.book.show.edit"), edit_book_review_path(review.book, review) %>
</div>
```

**Pull request merged**